services/121-service/src/transaction-job-processors/transaction-job-processors.service.ts

```diff
@@ -134,24 +145,20 @@ export class TransactionJobProcessorsService {
           )?.value as string, // This must be a string. If it is not, the intersolve API will return an error (maybe).
         });
     } catch (error) {
-      await this.createTransaction({
-        amount: input.transactionAmount,
-        registration: registration,
+      await this.processTransactionResult({
+        jobInput: input,
+        calculatedTranserAmount: input.transactionAmount, // TODO: STORE THE CALCULATED AMOUNT HERE AS THIS IS USED ON RETRY AND SHOWN IN PORTAL
```

Comment:
I think we need to adjust the flow to get this earlier

Reply:
See above comment.

If we are not talking about the multiplier here (which I believe we are not), but about the calculation that happens related to balance and spent this month, then I think we do not need to adjust the flow. It will just recalculate it on retry. Which is fine. Or perhaps even better.